### PR TITLE
Fix BBEs related to workers

### PR DESCRIPTION
**Pull request by NatashaWso2**

### CODE REVIEW

File: examples/async/async.bal [hunk 1]
```diff
@@ -40,28 +41,27 @@ public function main() {
     }
     io:println(f3.isDone());
 
-    // Asynchronously call the below functions and wait for either of them to finish
     future<int> f4 = start square(20);
     future<string> f5 = start greet("Bert");
-    // You can wait for either of the above asynchronous functions to finish
+    // You can wait for either of the asynchronous functions to finish.
+    // Here `f4` will finish before `f5` since a `runtime:sleep` is
```
Comment (pubudu91) [inline examples/async/async.bal]: shall we rephrase this to "since `runtim:sleep()` is called in the `greet()` function to delay its execution" or something?

File: examples/async/async.bal [hunk 2]
```diff
@@ -40,28 +41,27 @@ public function main() {
...
     io:println(f3.isDone());
 
-    // Asynchronously call the below functions and wait for either of them to finish
     future<int> f4 = start square(20);
     future<string> f5 = start greet("Bert");
-    // You can wait for either of the above asynchronous functions to finish
+    // You can wait for either of the asynchronous functions to finish.
+    // Here `f4` will finish before `f5` since a `runtime:sleep` is 
+    // added to the `greet()` function to delay its execution. The 
+    // asynchronous function that finishes first will be taken as the result.
```
Comment (pubudu91) [inline examples/async/async.bal]: Shall we modify this to something like "The value returned by the asynchronous function..."

File: examples/async/async.description [hunk 3]
```diff
@@ -1,3 +1,3 @@
 // Async functionality in Ballerina is a way to execute operations in an asynchronous manner by returning a `future`.
 // The `future` can be used to extract the result of the async function call (if available), check the running status of an execution,
-// or cancel the operation if needed. You can wait for a single or multiple async functions to complete using the 'wait' expression.
+// or cancel the operation if needed. You can wait for a single async function or multiple async functions to complete using the 'wait' expression.
```
Comment (pubudu91) [inline examples/async/async.description]: 'wait' -> `wait`

File: examples/fork-variable-access/fork_variable_access.bal [hunk 4]
```diff
@@ -14,49 +14,51 @@ public function main() {
         name , "] value of city is [", city, "] value of postcode is [",
         postcode, "]");
 
-    // Declare the fork statement.
+    // Declare the fork block.
     fork {
         worker W1 {
-            // Change the value of the integer variable `i` within the worker W1.
+            // Change value of the integer variable `i` within worker W1.
```
Comment (pubudu91) [inline examples/fork-variable-access/fork_variable_access.bal]: "the" should be there right?

File: examples/fork-variable-access/fork_variable_access.bal [hunk 5]
```diff
@@ -14,49 +14,51 @@ public function main() {
...
             i = 23;
-            // Change value of map variable `m` within the worker W1.
+            // Change value of the map variable `m` within worker W1.
```
Comment (pubudu91) [inline examples/fork-variable-access/fork_variable_access.bal]: W1 -> `W1`

Comment (pubudu91) [reply examples/fork-variable-access/fork_variable_access.bal]: same applies for other places as well

File: examples/fork-variable-access/fork_variable_access.description [hunk 6]
```diff
@@ -1,9 +1,8 @@
...
+// Workers within a fork can access the variables that are declared within the scope of that block.
 // The variables can be value type variables or reference type variables.
 // Value type variables such as `string`, `int`, `float`, and `byte` are passed into the workers as a copy
 // of the original data. Therefore, you are able to update the value of it within the worker and that value is only
 // defined within the specific worker.
 // Reference type variables such as `map`, `json`, `xml`, and `record` are passed into these workers as a
 // reference to the original data. Therefore, if you update the value of these variables within a worker, the value gets
-// updated for the entire function.
-// The `wait` expression will wait until all the defined workers are completed
+// updated for the entire function. You may wait for the worker returns using the `wait` expression.
```
Comment (pubudu91) [inline examples/fork-variable-access/fork_variable_access.description]: Shall we rephrase this to "You may wait for the worker to return using..."

File: examples/fork/fork.description [hunk 7]
```diff
@@ -0,0 +1,5 @@
+// In a function execution, a fork block splits the current execution to multiple workers, which do their work in parallel.
+// The `fork` block allows you to start any number of parallel workers at once. Workers are visible outside the fork block as futures. 
+// These futures behave similar to the futures returned by asynchronous functions created by `start`. You may wait for the worker 
+// returns using `wait` expression.
```
Comment (pubudu91) [inline examples/fork/fork.description]: using the*

File: examples/fork/fork.description [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+// You may also send `anydata` variables form one fork worker to another worker in the same fork.
```
Comment (pubudu91) [inline examples/fork/fork.description]: form -> from

File: examples/workers/tests/worker_test.bal [hunk 9]
```diff
@@ -11,7 +11,7 @@ int counter = 0;
 }
 public function mockPrint(any... s) {
     string outStr;
-    foreach str in s{
+    foreach var str in s{
```
Comment (pubudu91) [inline examples/workers/tests/worker_test.bal]: s{ -> s {

File: examples/workers/workers.bal [hunk 10]
```diff
@@ -1,9 +1,11 @@
 import ballerina/io;
 
-// In Ballerina, each function consists of one or more workers, which are independent 
-// parallel execution code blocks. If explicit workers are not mentioned with worker blocks,
-// the function code will belong to a single implicit default worker.
+// In Ballerina, each function consists of one or more workers, which are
```
Comment (lankavitharana) [inline examples/workers/workers.bal]: shall we say something about strands here?
